### ISSUE TITLE
New logic for the app

### ISSUE THREAD

**Issue by skabbes** [Merovingian87/Mindfulness-HTML5-App#1]
I noticed that it is quite buggy to use this app when you are pressing buttons while the timer is running.  If you simply change around some of the logic you are using, you will notice that doesn't happen anymore.

Instead of trying to always "remember" how much time has pass, just remember what time you pressed the start button, and just calculate it every time.

```javascript
let startTime;
let duration = 120;

function onMoreDuration() {
  duration = duration + 120;
}

function onLessDuration() {
  duration = duration - 120;
  if (duration < 0) {
    duration = 0;
  }
}

function onStartPress() {
    startTime = Date.now();
}

function every1Second() {
  let now = Date.now();
  let secondsPassed = (now - startTime) / 1000;
  let timeLeft = duration - secondsPassed;
  let minutes = Math.floor(timeLeft / 60);
  let seconds = timeLeft - (minutes * 60);
  console.log(minutes + ':' + seconds);
}
```

**Comment by Merovingian87** [comment]
Hey Steven, 

Thank you so much for your help! 

I've implemented the new logic and added a state-switch to make sure the "Start Session" button does not re-trigger the function while running. Same for the "+/-"  buttons. 

Also, I discovered how to use noSleep.js to avoid the screensaver and this makes the app usable, even when meditating for 60+ minutes!! 

Finally, I added the favicons so that people can add it to their home-screens with a nice ZenCircle logo as a default. 

My next step is to learn how to cache the .js logic so that the app can run offline. 

I will keep you posted :)
Thanks again!
